R/vaccination.R

```diff
@@ -506,108 +506,140 @@ vaccine_schedule <- function(date, doses, n_doses = 2L) {
+    population_left <- priority_population
+    doses_given <- apply(doses, c(1, 2), sum)
 
-##' Helper function to create a vaccination schedule that covers data
-##' from the past and projects doses into the future based on the last
-##' week of vaccination (based on JCVI order using
-##' [vaccine_schedule_future]). This function is subject to change.
-##'
-##' @title Vaccination schedule using data and future
-##'
-##' @inheritParams vaccine_schedule_from_data
-##' @inheritParams vaccine_schedule_future
-##' @inheritParams vaccine_priority_population
-##'
-##' @param end_date The final day in the future to create a schedule
-##'   for. After this date the model will assume 0 vaccine doses given
-##'   so an overestimate is probably better than an underestimate.
-##'
-##' @return A [vaccine_schedule] object
-##' @export
-vaccine_schedule_data_future <- function(data, region, uptake, end_date,
-                                         mean_days_between_doses,
-                                         booster_daily_doses_value = NULL,
-                                         booster_proportion = rep(1L, 19)) {
-
-  priority_population <- vaccine_priority_population(region, uptake)
-  n_carehomes <- priority_population[18:19, 1]
-  schedule_past <- vaccine_schedule_from_data(data, n_carehomes)
-  ## then average out last 7 days and project forward by n days
-  ## future; we will probably change this to avoid backfill by taking
-  ## the days -14..-8
-  i <- utils::tail(seq_len(dim(schedule_past$doses)[[3]]), 7)
-  mean_doses_last <- sum(schedule_past$doses[, , i], na.rm = TRUE) / length(i)
-  end_date <- as_sircovid_date(end_date)
-  n_days_future <- end_date -
-    (schedule_past$date + dim(schedule_past$doses)[[3]]) + 1L
-  daily_doses_future <- rep(round(mean_doses_last), n_days_future)
-  vaccine_schedule_future(
-    schedule_past,
-    daily_doses_future,
-    mean_days_between_doses,
-    priority_population,
-    booster_daily_doses_value = booster_daily_doses_value,
-    booster_proportion = booster_proportion)
+    for (j in seq_len(dim(priority_population)[2])) {
+      vaccinated <- pmin(population_left[, j, ], doses_given)
+      population_left[, j, ] <- population_left[, j, ] - vaccinated
+      doses_given <- doses_given - vaccinated
+    }
+
+    n_groups <- dim(population_left)[1]
+    n_priority_groups <- dim(population_left)[2]
+    n_doses <- dim(population_left)[3]
+    n_days <- dim(agg_doses)[2]
+
+    population_to_vaccinate_mat <- array(0, c(n_groups, n_priority_groups,
+                                              n_doses, n_days))
+    daily_doses_tt <- array(0, dim(agg_doses))
+
+    f <- function(dose) {
+      vaccination_schedule_exec(daily_doses_tt, agg_doses[dose, ],
+                                population_left,
+                                population_to_vaccinate_mat,
+                                Inf,
+                                0L, dose)
+    }
+
+    doses <- doses +
```

Comment:
I never know what this Reduce thing does and haven't checked the indexes of the apply but assuming this is ok

Reply:
reduces a list to a single value (by summing the items in the list in this case)